### DISCUSSION TITLE
override extra plugins configuration inside import spes

### DISCUSSION THREAD

**Discussion by dmshvetsov** [LazyVim/LazyVim#2534] [Q&A]
Following this [Lazy.nvim import feature](https://github.com/folke/lazy.nvim?tab=readme-ov-file#%EF%B8%8F-importing-specs-config--opts) 

> When you import specs, you can override them by simply adding a spec for the same plugin to your local specs, adding any keys you want to override / merge.

I tried to override opts of aerial plugin with the following config

```lua
-- lua/config/laze.lua
require("lazy").setup({
  spec = {
    -- add LazyVim and import its plugins
    { "LazyVim/LazyVim", import = "lazyvim.plugins" },
    -- import any extras modules here
    { import = "lazyvim.plugins.extras.editor.aerial", opts = { layout = { default_direction = "left" } } },
    -- import/override with your plugins
    { import = "plugins" },
  },
  -- ...rest omitted
}
```

and it didn't work

although the same opts table in the plugin folder did work

```lua
-- lua/plugins/override.lua
return {
  {
    "stevearc/aerial.nvim",
    opts = {
      layout = {
        default_direction = "left",
      },
    },
  },
}
```

Where did I make a mistake? Or do I not understand how override feature works in specs import?

**Comment by dpetka2001** [comment]
You can't do that in the `spec` field of `lazy.lua`. 

According to `lazy.nvim`'s [docs](https://github.com/folke/lazy.nvim/blob/main/lua/lazy/core/config.lua#L18) `spec` is of type `LazySpec`. [Here](https://github.com/folke/lazy.nvim/blob/aedcd79811d491b60d0a6577a9c1701063c2a609/lua/lazy/types.lua#L78) you can see what `LazySpec` can be. In the case of `import` it is a `LazySpecImport` and it only accepts these [values](https://github.com/folke/lazy.nvim/blob/aedcd79811d491b60d0a6577a9c1701063c2a609/lua/lazy/types.lua#L80-L83), namely `enabled` and `cond`. 

So, the correct way of altering a plugin spec would be like you did in your second example by altering it in a file under `/lua/plugins/` directory.
